File: friend-impl/src/main/java/sample/chirper/friend/impl/FriendModule.java
/*
 * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
 */
package sample.chirper.friend.impl;

import com.google.inject.AbstractModule;
import sample.chirper.friend.api.FriendService;
import com.lightbend.lagom.javadsl.server.ServiceGuiceSupport;

public class FriendModule extends AbstractModule implements ServiceGuiceSupport {
  @Override
  protected void configure() {
    bindServices(serviceBinding(FriendService.class, FriendServiceImpl.class));
  }
}

File: friend-impl/src/main/scala/sample/chirper/friend/impl/FriendModule.scala
/*
 * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
 */
package sample.chirper.friend.impl

import com.google.inject.AbstractModule
import com.lightbend.lagom.javadsl.server.ServiceGuiceSupport
import sample.chirper.friend.api.FriendService

class FriendModule extends AbstractModule with ServiceGuiceSupport {
  override protected def configure(): Unit = {
    bindServices(serviceBinding(classOf[FriendService], classOf[FriendServiceImpl]));
  }
}
